sway/tree/output.c

```diff
@@ -195,18 +195,25 @@ static void output_evacuate(struct sway_output *output) {
 			new_output = root->noop_output;
 		}
 
-		if (workspace_is_empty(workspace)) {
-			// If floating is not empty, there are sticky containers to move
-			if (workspace->floating->length) {
+		struct sway_workspace *new_output_ws =
+			output_get_active_workspace(new_output);
+
+		// workspace_is_empty() only checks that no tiling containers and no
+		// non-sticky floating containers exist
+		bool ws_is_empty = workspace_is_empty(workspace);
+
+		// If the workspace has only sticky floating containers and the new
+		// output has an active workspace (the noop output may not have one),
+		// move the containers to it.
+		if (ws_is_empty && workspace->floating->length && new_output_ws) {
```

Comment:
It's a bit nonobvious (though correct) that if `ws_is_empty` is true here and `workspace->floating->length` is nonzero, then all of those floating views are sticky views.

I think introducing a `workspace_num_sticky_views` (to complement `workspace_num_tiling_views`) in `workspace.c` and using that here would make the intent more obvious.